util/scale_run.sh

```diff
@@ -0,0 +1,330 @@
+    echo "  -i|--run-id           Set a value to prepend to scale run directories"
+    echo "                        (default: '')"
+    echo "  --[no]tune            apply pe-tune to deployment"
+    echo "                        (default: 'true')"
+    echo
+}
+
+# Debug function to validate that bash subshells inherit
+# expected environment variables.
+function echo_env () {
+    echo "============================================
+Provided args:
+TUNE=$TUNE
+RUN_ID=$RUN_ID
+PE_VERSION=$PE_VERSION
+
+In subshell for $test $run_type round $i
+cmd=$cmd
+BEAKER_INSTALL_TYPE=$BEAKER_INSTALL_TYPE
+BEAKER_PE_DIR=$BEAKER_PE_DIR
+BEAKER_PE_VER=$BEAKER_PE_VER
+PUPPET_GATLING_SCALE_BASE_INSTANCES=$PUPPET_GATLING_SCALE_BASE_INSTANCES
+PUPPET_GATLING_SCALE_ITERATIONS=$PUPPET_GATLING_SCALE_ITERATIONS
+PUPPET_GATLING_SCALE_INCREMENT=$PUPPET_GATLING_SCALE_INCREMENT
+PUPPET_GATLING_SCALE_SCENARIO=$PUPPET_GATLING_SCALE_SCENARIO
+ABS_AWS_MOM_SIZE=$ABS_AWS_MOM_SIZE
+============================================"
+}
+
+# Prepare checkout of puppet-gatling-load-test
+function prep_gplt () {
+    mkdir -p "$WORK_DIR"
+    cd "$WORK_DIR" || exit 1
+    git clone git@github.com:puppetlabs/gatling-puppet-load-test "$1"
+    cd "$1" || exit 1
+    bundle install --path vendor/bundle
+}
+
+
+##########
+# Argument checking
+##########
+
+ARGS=()
+while [ $# -gt 0 ]
+do
+    case "$1" in
+        -h|--help)  usage
+            exit 1
+            ;;
+        -d|--debug)
+            DEBUG=true
+            shift
+            ;;
+        -i|--run-id)
+          RUN_ID=$2
+          PREFIX="$PREFIX-$RUN_ID"
+          shift 2
+          ;;
+        --tune)
+          TUNE=true
+          shift
+          ;;
+        --notune)
+          TUNE=false
+          shift
+          ;;
+        --) # end argument parsing
+            shift
+            break
+            ;;
+        -*|--*=) # unsupported flags
+            echo "Error: Unsupported flag $1" >&2
+            exit 1
+            ;;
+        *) # preserve positional arguments
+            ARGS+=("$1")
+            shift
+            ;;
+        esac
+done
+
+# Ensure that TUNE is set to true by default
+TUNE="${TUNE:-true}"
+
+# Ensure PE_VERSION is provided
+if [ "${#ARGS[@]}" -le "0" ]; then
+    echo "PE_VERSION not provided"
+    echo
+    usage
+    exit 1
+fi
+PE_VERSION=${ARGS[0]}
+
+# Ensure PE_VERSION is semver
+# e.g. 2019.1.1-rc0-207-g4e47830
+# Regex from: https://regexr.com/39s32
+RE='[^0-9]*\([0-9]*\)[.]\([0-9]*\)[.]\([0-9]*\)\([0-9A-Za-z-]*\)'
+PE_MAJOR=$(echo "$PE_VERSION" | sed -e "s#$RE#\1#")
+PE_MINOR=$(echo "$PE_VERSION" | sed -e "s#$RE#\2#")
+PE_PATCH=$(echo "$PE_VERSION" | sed -e "s#$RE#\3#")
+PE_BUILD=$(echo "$PE_VERSION" | sed -e "s#$RE#\4#")
+echo "PE_MAJOR=$PE_MAJOR"
+echo "PE_MINOR=$PE_MINOR"
+echo "PE_PATCH=$PE_PATCH"
+echo "PE_BUILD=$PE_BUILD"
+#if ! [[ $PE_VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+((-[0-9a-zA-Z-]+)*)?$ ]]; then
```

Comment:
if intentionally leaving this here, comment on why.

Reply:
This line has been deleted